packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,195 @@
+        `No "dist/server/" folder found in ${distServerPath}. Ensure the app.json key "expo.web.output" is set to "server"`
+      );
+    }
+
+    async function* emitWorkerTarballAsync(
+      assetMap: WorkerAssets.AssetMap
+    ): AsyncGenerator<WorkerAssets.FileEntry> {
+      yield ['assets.json', JSON.stringify(assetMap)];
+
+      // TODO: Create manifest from user configuration
+      const manifest = { env: {} };
+      yield ['manifest.json', JSON.stringify(manifest)];
+
+      const workerFiles = WorkerAssets.listWorkerFiles(distServerPath);
+      for await (const workerFile of workerFiles) {
+        yield [`server/${workerFile.normalizedPath}`, workerFile.data];
+      }
+    }
+
+    async function uploadTarballAsync(tarPath: string): Promise<any> {
+      const uploadUrl = await getSignedDeploymentUrlAsync(graphqlClient, exp, {
+        appId: projectId,
+      });
+
+      const { response } = await uploadAsync({
+        url: uploadUrl,
+        filePath: tarPath,
+        compress: false,
+        headers: {
+          accept: 'application/json',
+        },
+      });
+      if (response.status === 413) {
+        throw new Error(
+          'Upload failed! (Payload too large)\n' +
+            `The gzipped files in "dist/server/" (at: ${distServerPath}) exceed the maximum file size.`
+        );
+      } else if (!response.ok) {
+        throw new Error(`Upload failed! (${response.statusText})`);
+      } else {
+        const json = await response.json();
+        if (!json.success || !json.result || typeof json.result !== 'object') {
+          throw new Error(json.message ? `Upload failed: ${json.message}` : 'Upload failed!');
+        }
+        return json.result;
+      }
+    }
+
+    async function uploadAssetsAsync(
+      assetMap: WorkerAssets.AssetMap,
+      uploads: Record<string, string>
+    ): Promise<void> {
+      if (typeof uploads !== 'object' || !uploads) {
+        return;
+      }
+
+      // TODO(@kitten): Batch and upload multiple files in parallel
+      const uploadParams: UploadParams[] = [];
+      for await (const asset of WorkerAssets.listAssetMapFiles(distClientPath, assetMap)) {
+        const uploadURL = uploads[asset.normalizedPath];
+        if (uploadURL) {
+          uploadParams.push({ url: uploadURL, filePath: asset.path });
+        }
+      }
+
+      const progress = {
+        total: uploadParams.length,
+        pending: 0,
+        percent: 0,
+        transferred: 0,
+      };
+
+      const updateProgress = createProgressTracker({
+        total: progress.total,
+        message(ratio) {
+          const percent = `${Math.floor(ratio * 100)}`;
+          const details = chalk.dim(
+            `(${progress.pending} Pending, ${progress.transferred} Completed, ${progress.total} Total)`
+          );
+          return `Uploading client assets: ${percent.padStart(3)}% ${details}`;
+        },
+        completedMessage: 'Uploaded client assets for worker deployment',
```

Comment:
```suggestion
        completedMessage: 'Uploaded assets for serverless deployment',
```

Not sure if it's nice to "leak" this implementation detail to the user. Since we name it "serverless" on our dashboard, might be good to use the same naming.

There are a couple more instances, e.g. through `ora` messages. Might be nice to clean these up too?

Reply:
I'll have a look 👍 and will go through them again